tails_files/install.sh

```diff
@@ -68,23 +87,360 @@ if [ -f $XSESSION_RC ]; then
+    "${tails_live_dotfiles}/.local/share/applications/source.desktop" \
+    "${tails_live_dotfiles}/Desktop/document.desktop" \
+    "${tails_live_dotfiles}/Desktop/source.desktop"
+
+  chmod 700 \
+    "${amnesia_desktop}/document.desktop" \
+    "${amnesia_desktop}/source.desktop" \
+    "${amnesia_home}/.local/share/applications/document.desktop" \
+    "${amnesia_home}/.local/share/applications/source.desktop" \
+    "${tails_live_dotfiles}/.local/share/applications/document.desktop" \
+    "${tails_live_dotfiles}/.local/share/applications/source.desktop" \
+    "${tails_live_dotfiles}/Desktop/document.desktop" \
+    "${tails_live_dotfiles}/Desktop/source.desktop"
+}
+
+function configure_network_manager_hook()
+{
+  # Install a persistent NetworkManager hook to write ATHS values to the
+  # system /etc/tor/torrc, so the Document Interface can be accessed via
+  # Tor Browser. On the Admin Workstation, the SSH ATHS values will also
+  # be added. The hook will run on "up" for any interface other than loopback,
+  # and is carefully ordered to run AFTER the Tails Tor hook and BEFORE the
+  # Tails Apt update hook.
+  if ! grep -q 'custom-nm-hooks' "${tails_live_persistence}/persistence.conf"; then
+    echo "/etc/NetworkManager/dispatcher.d	source=custom-nm-hooks,link" >> "${tails_live_persistence}/persistence.conf"
+  fi
+  mkdir -p "${tails_live_persistence}/custom-nm-hooks"
+
+  cp -f 65-configure-tor-for-securedrop.sh "${tails_live_persistence}/custom-nm-hooks"
+  cp -f 65-configure-tor-for-securedrop.sh "$network_manager_dispatcher"
+  chown root:root \
+    "${tails_live_persistence}/custom-nm-hooks/65-configure-tor-for-securedrop.sh" \
+    "${network_manager_dispatcher}/65-configure-tor-for-securedrop.sh"
+  chmod 755 \
+    "${tails_live_persistence}/custom-nm-hooks/65-configure-tor-for-securedrop.sh" \
+    "${network_manager_dispatcher}/65-configure-tor-for-securedrop.sh"
+
+  # Run the SecureDrop init script. Among other things, implements the torrc
+  # additions configured by this script. See `securedrop_init.py` for details.
+  /usr/bin/python "${securedrop_dotfiles}/securedrop_init.py"
+}
+
+function print_success_message()
+{
+  cat <<SD_COMPLETE_MSG1
+Successfully configured Tor and set up desktop bookmarks for SecureDrop!
+You will see a notification appear on your screen when Tor is ready.
+
+The Document Interface's Tor onion URL is: http://$(lookup_document_aths_url)
+The Source Interfaces's Tor onion URL is: http://$(lookup_source_ths_url)
+SD_COMPLETE_MSG1
+
+  # We're assuming the SSH aliases have been set up as long as ~/.ssh/config exists.
+  # Intentionally padding with an empty line to improve readability.
+  if is_admin_workstation; then
+    cat <<SD_COMPLETE_MSG2
+
+The App Server's SSH hidden service address is: $(lookup_app_ssh_aths_url)
+The Monitor Server's SSH hidden service address is: $(lookup_mon_ssh_aths_url)
+SSH aliases are set up. You can use them with 'ssh app' and 'ssh mon'.
```

Comment:
Since the conditional is upon `if_admin_workstation`, this message about the aliases could still be output (as misleading) even if per line https://github.com/freedomofpress/securedrop/pull/1309/files#diff-a44dd5740ffe7a0588a02b9c152ec8edR255 there's an existing SSH config. I think you probably already know that though and made a judgment call that it's okay. That would only happen if an old, customized Tails had been re-purposed for SecureDrop, so it's not likely.